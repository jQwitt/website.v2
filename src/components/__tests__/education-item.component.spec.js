import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { EducationItem } from '../index';

afterEach(cleanup);
describe('EducationItem renders correctly', () => {
  test('snapshot', () => {
    const { asFragment } = render(<EducationItem />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('conditionally renders title', () => {
    const data = {
      title: 'Title',
    };
    const { getByText } = render(<EducationItem title={data.title} />);
    expect(getByText(data.title)).toBeTruthy();
  });
  it('conditionally renders subTitle', () => {
    const data = {
      subTitle: 'subTitle',
    };
    const { getByText } = render(<EducationItem subTitle={data.subTitle} />);
    expect(getByText(data.subTitle)).toBeTruthy();
  });
  it('conditionally renders description', () => {
    const data = {
      description: 'description',
    };
    const { getByText } = render(
      <EducationItem description={data.description} />
    );
    expect(getByText(data.description)).toBeTruthy();
  });

  it('renders all props correctly', () => {
    const data = {
      title: 'Title',
      subTitle: 'subTitle',
      description: 'description',
    };
    const { getByText } = render(
      <EducationItem
        title={data.title}
        subTitle={data.subTitle}
        description={data.description}
      />
    );
    expect(getByText(data.title)).toBeTruthy();
    expect(getByText(data.subTitle)).toBeTruthy();
    expect(getByText(data.description)).toBeTruthy();
  });
});
