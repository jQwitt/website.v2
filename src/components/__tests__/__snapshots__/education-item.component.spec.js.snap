// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationItem renders correctly snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-xl-6 col-lg-6 col-md-6 col-sm-6 col-12 education-item"
  >
    <div
      class="row justify-content-start"
    />
    <div
      class="row"
    />
  </div>
</DocumentFragment>
`;
